src/main/java/filter/expression/Qualifier.java

```diff
@@ -136,23 +131,61 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
+    /**
+     * Get all milestones that can be aliased with the keyword "current".
+     */
+    private static List<TurboMilestone> getMilestonesWithAlias(IModel model, List<String> repoIds) {
+        List<TurboMilestone> milestones = new ArrayList<>();
+        // add milestones with due date first
+        milestones.addAll(model.getMilestones().stream()
+                .filter(ms -> ms.getDueDate().isPresent())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .sorted((a, b) -> getMilestoneDueDateComparator().compare(a, b))
+                .collect(Collectors.toList()));
+
+        // Special case: if there is only one open milestone, it should be included regardless of whether
+        // it has due date or not.
+        // In this case, if it is not included already (the open milestone does not
+        // have due date), append it to the milestone list.
+        List<TurboMilestone> openMilestones = getOpenMilestones(model, repoIds);
+        if (openMilestones.size() == 1) {
+            TurboMilestone openMilestone = openMilestones.get(0);
+            boolean hasDueDate = openMilestone.getDueDate().isPresent();
+            if (!hasDueDate) {
+                milestones.add(openMilestone);
+            }
+        }
+
+        return milestones;
+    }
+
+    private static List<TurboMilestone> getOpenMilestones(IModel model, List<String> repoIds) {
+        return model.getMilestones().stream()
+                .filter(ms -> ms.isOpen())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .collect(Collectors.toList());
+    }
+
     private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
         if (allMilestones.isEmpty()) {
             return Optional.empty();
         }
 
-        int currentIndex = 0;
-
-        for (TurboMilestone checker : allMilestones) {
-            boolean overdue = checker.getDueDate().isPresent() &&
-                    checker.getDueDate().get().isBefore(LocalDate.now());
-            boolean relevant = !(overdue && checker.getOpenIssues() == 0);
+        Optional<TurboMilestone> openAndRelevantMilestone = allMilestones.stream()
+                .filter(ms -> ms.isOpen() && isRelevantMilestone(ms))
+                .findFirst();
 
-            if (checker.isOpen() && relevant) {
-                return Optional.of(currentIndex);
+        if (openAndRelevantMilestone.isPresent()) {
+            return Optional.of(allMilestones.indexOf(openAndRelevantMilestone.get()));
+        } else {
```

Comment:
The `else` case here seems to cover only when there is 1 open milestone with no due date. If that is the case, we can put this as a case above after `if (allMilestones.isEmpty())`. Something along `if (allMilestones.stream().filter(ms ->ms.isOpen()).count() == 1)`. The problem with this would be an open milestone that is forgotten to be closed (i.e. overdue but has no open issues in it) can be considered `current`.

If you would like to be more specific, since we would like to cover the case where `current` milestone is the only open milestone with no due date, our `if` case can be changed to check if there is any milestone with no due date in the list; if yes, return that as `current`. We can do that __if__ we expect the list to have at most 1 milestone with no due date. This makes the method to be overly specific but it works for our case.